### ISSUE TITLE
Allow calling shell to pass environment variables to Actions environment

### ISSUE THREAD

**Issue by mtribiere** [bazelbuild/bazel#18586]
### Description of the feature request:

Currently, the Actions  environment working behavior is a little odd, as there is no options to forward all environment variables to the Action environment. Just `LD_LIBRARY_PATH` and `PATH` are forwarded but there should be an options to forward them all.

This would be very useful in the embedded world, and old test benches who's configurations can't be changed.

### What underlying problem are you trying to solve with this feature?

Allow old compilers and tools to benefits from Bazel. But those are picky and require their environment variables. 

### Which operating system are you running Bazel on?

Windows

### What is the output of `bazel info release`?

release 6.2.0

### Have you found anything relevant by searching the web?

https://github.com/bazelbuild/bazel/issues/8578
https://groups.google.com/g/bazel-discuss/c/AZek-OR1t7A

**Comment by fmeum** [comment]
You can forward environment variables individually via `--action_env=VAR_NAME`. Rules may choose to ignore the value, but the built-in ones don't.

If *all* variables were made available to an action, caching would probably become impossible. Is that really what you want? Is it possible to enumerate the variables consumed by the tools you are using?

**Comment by mtribiere** [comment]
@fmeum Thanks for your quick answer.

The way it works right now is the following :
- We have a `.bat` file that generate a bunch of environment variables based on different conditions
- Those variables are exported to windows with `set MYVAR=VALUE`
- Bazel is called to start the compilation

The problem right now, is that all those environment variables are accessible in the windows shell, and I cannot forward them all one by one to Bazel with  `--action_env=VAR_NAME`. The solution to me would be an option to allow the forwarding of all  environment variables, but I'm open to any suggestion you may have.

Thanks

**Comment by mtribiere** [comment]
I didn't made that very clear, my bad! Those variables are generated, including their names. There is no way to know which one will be produced in advanced, as this is based on a number of external factors.

**Comment by mtribiere** [comment]
Exactly, the variables are generated at a custom configuration stage, before any compilation tool is called. Make sense, maybe an external tool would to the trick... Caching is indeed important for us.

**Comment by fmeum** [comment]
There is no way to do this as far as I know, mostly because features that would essentially prevent caching aren't really going to be considered.

Glad to hear that there is a path forward for you though.
